app/helpers/apply-line-break.js

```diff
@@ -1,12 +1,7 @@
 import Ember from "ember";
 
 export default Ember.Helper.helper(function(value) {
-  var text;
-  if(/<[a-z][\s\S]*>/i.test(value)) {
-    text = value;
-  } else {
-    text = Ember.Handlebars.Utils.escapeExpression(value);
```

Comment:
Why are we removing `escapeExpression` here?

Reading https://guides.emberjs.com/v2.13.0/templates/writing-helpers/ confirms "make sure you escape anything that may have come from an untrusted user with the escapeExpression utility"

SafeString below is simply creating a string that we are already claiming is safe and turns off all of the html escaping we need. Is there another fragment I'm missing, that does this?

cc @patrixr

Reply:
We should escape it all the time, shouldn't we ?

```javascript
return new Ember.String.htmlSafe(
  Ember.Handlebars.Utils.escapeExpression(value.replace(/(\r\n|\n|\r)/gm, "<br>"));
)
```

Reply:
We escape expression [here](https://github.com/crossroads/shared.goodcity/blob/master/app/services/message-link-convertor.js#L10) before saving messages and that is the reason I have not added it at the helper because it will then convert all symbolic characters to HTML
cc: @steveyken  @patrixr

Reply:
@shreyas098 yes, good to escape at input time but output is also 'unsafe' - imagine the scenario where db has been compromised... js could be inserted which would get to the client app un-sanitized. need to do it always on output, regardless of input.

Reply:
sure @steveyken . Change is added